### PR TITLE
[Docs] Add tentacle package development

### PR DESCRIPTION
**Pull request by Herklos**
image

image

image

### CODE REVIEW

File: docs/source/pages/Tentacle-Package-Development.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    tentacles/YOUR_TP_CATEGORY/YOUR_TP_SUB_CATEGORY/YOUR_TENTACLE_PACKAGE_NAME/metadata.json
+
+A tentacle package metadata.json contains :
+
+.. code-block:: json
+
+    {
+      "version": "YOUR_TP_VERSION",
+      "origin_package": "YOUR_TP_ORIGIN_PACKAGE",
+      "tentacles": [YOUR_TP_TENTACLE_1, YOUR_TP_TENTACLE_2],
```
Comment (GuillaumeDSM) [inline docs/source/pages/Tentacle-Package-Development.rst:43]:
```suggestion
      "tentacles": ["YOUR_TP_TENTACLE_1", "YOUR_TP_TENTACLE_2"],
```

File: docs/source/pages/Tentacle-Package-Development.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      "tentacles-requirements": [YOUR_TP_TP_REQUIREMENT_1, YOUR_TP_TP_REQUIREMENT_2]
```
Comment (GuillaumeDSM) [inline docs/source/pages/Tentacle-Package-Development.rst:44]:
```suggestion
      "tentacles-requirements": ["YOUR_TP_TP_REQUIREMENT_1", "YOUR_TP_TP_REQUIREMENT_2"]
```

File: docs/source/pages/Tentacle-Package-Development.rst [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+.. NOTE:: TP is for tentacle package
+
+- **YOUR_TP_CATEGORY** can be Backtesting, Evaluator, Services or Trading
+- **YOUR_TP_SUB_CATEGORY** should be a sub category of **YOUR_TP_CATEGORY** in the existing tentacle architecture
+- **YOUR_TENTACLE_PACKAGE_NAME** is the name of your tentacle package, shouldn't use an existing tentacle package name
+
+Description file
+^^^^^^^^^^
+
+A tentacle package contains metadata described in metadata.json file.
```
Comment (GuillaumeDSM) [inline docs/source/pages/Tentacle-Package-Development.rst:29]:
```suggestion
A tentacle package contains metadata described in metadata.json file. This file is used to properly install the tentacle and should be carefully written.
```

File: docs/source/pages/Tentacle-Package-Development.rst [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      "tentacles": [YOUR_TP_TENTACLE_1, YOUR_TP_TENTACLE_2],
+      "tentacles-requirements": [YOUR_TP_TP_REQUIREMENT_1, YOUR_TP_TP_REQUIREMENT_2]
+    }
+
+
+- **YOUR_TP_VERSION** is your tentacle package version
+- **YOUR_TP_ORIGIN_PACKAGE** is the author or the origin repository of the tentacle package
+- **YOUR_TP_TENTACLE_1** and **YOUR_TP_TENTACLE_2** are names of your tentacle package tentacles (can have 1 or more).
+- **YOUR_TP_TP_REQUIREMENT_1** and **YOUR_TP_TP_REQUIREMENT_2** are the names of required tentacle packages to have installed to run your tentacle package (can have 0 or more)
+.. NOTE:: **YOUR_TP_TENTACLE_X** should match the tentacle python class name
```
Comment (GuillaumeDSM) [inline docs/source/pages/Tentacle-Package-Development.rst:52]:
```suggestion
.. NOTE:: **YOUR_TP_TENTACLE_X** should match python classes to be exposed in the tentacle
```

File: docs/source/pages/Tentacle-Package-Development.rst [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+Config
+^^^^^^^^^^
+
+A tentacle package can contain tentacle config. Config files are located in *config/* folder at :
+
+.. code-block:: shell
+
+    tentacles/YOUR_TP_CATEGORY/YOUR_TP_SUB_CATEGORY/YOUR_TENTACLE_PACKAGE_NAME/config/
+
+Each tentacles config file should be named with the exact case and name as the tentacle.
```
Comment (GuillaumeDSM) [inline docs/source/pages/Tentacle-Package-Development.rst:96]:
```suggestion
Each tentacles config file should be named with the exact case and name as the exposed tentacle class(es).```

File: docs/source/pages/Tentacle-Package-Development.rst [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+Below an example for *MyAwesomeTentacle* :
+
+.. code-block:: shell
+
+    tentacles/YOUR_TP_CATEGORY/YOUR_TP_SUB_CATEGORY/YOUR_TENTACLE_PACKAGE_NAME/config/MyAwesomeTentacle.json
+
+Same for tentacles config schema file. Below an example for *MyAwesomeTentacle* :
```
Comment (GuillaumeDSM) [inline docs/source/pages/Tentacle-Package-Development.rst:103]:
```suggestion
Tentacle configuration interface are generated using configuration schema files and the `json-editor <https://github.com/json-editor/json-editor>`_ library.
For the configuration interface to work, schema files have to be named like their associated configuration files with the *_schema* suffix. Here is an example for *MyAwesomeTentacle* :
```

**Review by GuillaumeDSM — COMMENTED** [review]
great !

File: docs/source/pages/Tentacle-Package-Development.rst (hunk 6 above)
Comment (Herklos) [reply docs/source/pages/Tentacle-Package-Development.rst:103]: :+1:

**Review by GuillaumeDSM — APPROVED** [review]

**Pull request merged**